acpi/src/platform/mod.rs

```diff
@@ -132,3 +132,55 @@ where
         Ok(PlatformInfo { power_profile, interrupt_model, processor_info, pm_timer })
     }
 }
+
+/// Wakes up Application Processors (APs) using the Multiprocessor Wakeup Mailbox mechanism.
+///
+/// This function writes to a specific memory-mapped mailbox to signal APs to wake up and start
+/// executing from a given wakeup vector. It waits for the APs to acknowledge the wakeup command
+/// by monitoring the mailbox command field until it is reset to `Noop`.
+pub fn wakeup_aps<H>(
+    tables: &AcpiTables<H>,
+    handler: H,
+    apic_id: u32,
+    wakeup_vector: u64,
+    timeout_loops: u64,
+) -> Result<(), AcpiError>
+where
+    H: AcpiHandler,
+{
+    let madt = tables.find_table::<Madt>()?;
+    let mailbox_addr = madt.get_mpwk_mailbox_addr()?;
+    let mut mpwk_mapping = unsafe {
+        handler.map_physical_region::<MultiprocessorWakeupMailbox>(
+            mailbox_addr as usize,
+            mem::size_of::<MultiprocessorWakeupMailbox>(),
+        )
+    };
+
+    // Reset command
+    mpwk_mapping.command = MpProtectedModeWakeupCommand::Noop as u16;
```

Comment:
I've not tested the mailbox myself, or verified via assembly, but I think these writes should also be volatile as Rust won't understand the necessity to write to the command field twice. Even if this produced correct assembly at the moment, we shouldn't rely on future compilers to.